beets/ui/commands.py

```diff
@@ -821,6 +471,361 @@ def should_resume(self, path):
+            else:
+                cur_title = item.title.strip()
+                lhs, rhs = ui.colordiff(cur_title, new_title)
+            lhs_width = len(cur_title)
+
+            # Track number change.
+            cur_track, new_track = format_index(item), format_index(track_info)
+            if cur_track != new_track:
+                if item.track in (track_info.index, track_info.medium_index):
+                    color = 'text_highlight_minor'
+                else:
+                    color = 'text_highlight'
+                templ = ui.colorize(color, u' (#{0})')
+                lhs += templ.format(cur_track)
+                rhs += templ.format(new_track)
+                lhs_width += len(cur_track) + 4
+
+            # Length change.
+            if item.length and track_info.length and \
+                    abs(item.length - track_info.length) > \
+                    config['ui']['length_diff_thresh'].as_number():
+                cur_length = ui.human_seconds_short(item.length)
+                new_length = ui.human_seconds_short(track_info.length)
+                templ = ui.colorize('text_highlight', u' ({0})')
+                lhs += templ.format(cur_length)
+                rhs += templ.format(new_length)
+                lhs_width += len(cur_length) + 3
+
+            # Penalties.
+            penalties = penalty_string(match.distance.tracks[track_info])
+            if penalties:
+                rhs += ' %s' % penalties
+
+            if lhs != rhs:
+                lines.append((' * %s' % lhs, rhs, lhs_width))
+            elif config['import']['detail']:
+                lines.append((' * %s' % lhs, '', lhs_width))
+
+        # Print each track in two columns, or across two lines.
+        col_width = (ui.term_width() - len(''.join([' * ', ' -> ']))) // 2
+        if lines:
+            max_width = max(w for _, _, w in lines)
+            for lhs, rhs, lhs_width in lines:
+                if not rhs:
+                    print_(lhs)
+                elif max_width > col_width:
+                    print_(u'%s ->\n   %s' % (lhs, rhs))
+                else:
+                    pad = max_width - lhs_width
+                    print_(u'%s%s -> %s' % (lhs, ' ' * pad, rhs))
+
+        # Missing and unmatched tracks.
+        if match.extra_tracks:
+            print_('Missing tracks ({0}/{1} - {2:.1%}):'.format(
+                len(match.extra_tracks),
+                len(match.info.tracks),
+                len(match.extra_tracks) / len(match.info.tracks)
+                ))
+        for track_info in match.extra_tracks:
+            line = ' ! %s (#%s)' % (track_info.title, format_index(track_info))
+            if track_info.length:
+                line += ' (%s)' % ui.human_seconds_short(track_info.length)
+            print_(ui.colorize('text_warning', line))
+            self.logger.info("missing track: {0} from {1}".format(
+                track_info.title,
+                match.info.album,
+            ))
```

Comment:
Oops! My apologies for missing that obvious detail.

The other tricky thing here is that you seem to be logging the extra tracks when the match is *displayed*, not when it is applied. So if someone considers two alternative album matches for one album they have, they'll get two different log entries for this. I think this kind of logging needs to occur after duplicate resolution, etc.